CODEOWNERS

```diff
@@ -33,6 +33,7 @@ Kconfig*                                  @tejlmand
 # All subfolders
 /drivers/                                 @anangl
 /drivers/bt_ll_nrfxlib/                   @anangl @rugeGerritsen
+/drivers/ieee802154/                      @rlubos
```

Comment:
I'm wondering about the directory structure here, we'll end up with two modules, at the same level, one adding the configuration module, other the platform files. Probably we should integrate them? i.e. leave this PR as it is, and move `drivers/nrf_802154_sl` into `drivers/ieee802154/platform`? What do you think?

btw, Please consider adding yourself here as well, if you want to be automatically added to PRs as reviewer.

Reply:
Actually, we deliver two modules that are needed for the 802.15.4 driver: nrf_802154 (open-source) and nrf_802154_sl (closed-source). Each of these modules has separated platform implementation. As nrf_802154 is open-source and going to be a part of Zephyr, the platform implementation is in hal_nordic repository. But for nrf_802154_sl the platform implementation is in sdk-nrf/drivers/nrf_802154_sl. I would prefer to keep directories regarding these modules separated to avoid confusion if given platform implementation is for nrf_802154 or for nrf_802154_sl.
The configuration added with this PR is for nrf_802154. I could rename the directory to nrf_802154 if it should follow the prefix of the project it refers to, but I would prefer to keep it separated from nrf_802154_sl.

Reply:
Ok, understood 👍